### PR TITLE
Make hashing a SymInt raise an error again

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130548

See https://github.com/pytorch/pytorch/issues/130547

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by albanD — APPROVED** [review]
SGTM

**Comment by ezyang** [comment]
@clee2000 looking over the fail line, it seems more likely it's just a land race (the line in question doesn't exist on the base commit here)

**Comment by ezyang** [comment]
cc @rec for latest changes

**Review by lezcano — APPROVED** [review]
Sounds good.
For the failing test, feel free to xfail it and @rec will send a patch

File: torch/__init__.py [hunk 1]
```diff
@@ -511,24 +511,31 @@ def _sympy_(self):
...
+        return self, 1
 
-    def bit_length(self) -> "SymInt":
-        return SymInt(self.node.wrap_int(self._get_int().bit_length()))
+    def bit_length(self) -> builtins.int:
+        # TODO: A more relaxed guard is possible here, where you guard to
+        # allow all integer quantities which would result in the same bit
+        # length.  We can also just make a dedicated Sympy function for
+        # computing this quantity and represent it symbolically.
+        return builtins.int(self).bit_length()
```
Comment (lezcano) [inline torch/__init__.py:534]: oh, I thought that the bug was coming from somewhere else. I guess this should be
```suggestion
        v = self.node.nested_int() if self.node.is_nested_int() else builtins.int(self)
        return v.bit_length()
```
which is what `_get_int` was doing

Comment (rec) [reply torch/__init__.py:534]: I'm a little unclear if anything is now expected of me. :-)

Just ping me on this thread, or file an issue with my name on it, if further action is needed.

Comment (lezcano) [reply torch/__init__.py:534]: Nothing from you for now :)

**Pull request closed without merging**